articles/contributing/style-guide.md

```diff
@@ -0,0 +1,451 @@
+
+# [Examples](#tab/examples)
+
+|   | Name | Description |
+|---|------|-------------|
+| ☑ | `X` | Well-understood shorthand for "apply an $X$ transformation" |
+| ☑ | `CNOT` | Well-understood shorthand for "controlled-NOT" |
+| ☒ | <s>`Cnot`</s> | Shorthand should not be in `CamelCase`. |
+| ☑ | `ApplyQft` | Common initialism "QFT" appears as a part of a long-form name. |
+| ☑ | `QFT` | Common initialism "QFT" appears as a part of a shorthand name. |
+
+
+
+***
+
+
+### Proper Nouns in Names ###
+
+While in physics it is common to name things after the first person to publish about them, most non-physicists aren’t familiar with everyone’s names and all of the history.
+Relying too heavily on naming conventions from physics can thus put up a substantial barrier to entry, as users from other backgrounds must learn a large number of seemingly opaque names in order to use common operations and concepts.
+<!-- An important part of the task of reducing confusion is to make code more accessible.
+Especially in a field such as quantum computing that is rich with domain expertise, we must at all times be cognizant of the demands we place on that expertise as we design quantum software.
+In naming code symbols, one way that this cognizance expresses itself is as an awareness of the convention from physics of adopting as the names of algorithms and operations the names of their original publishers.
+While we must maintain the history and intellectual provenance of concepts in quantum computing, demanding that all users be versed in this history to use even the most basic of functions and operations places a barrier to entry that is in most cases severe enough to even present an ethical compromise. -->
+Thus, we recommend that wherever reasonable, common nouns that describe a concept be adopted in strong preference to proper nouns that describe the publication history of a concept.
+As a particular example, the singly controlled SWAP and doubly controlled NOT operations are often called the "Fredkin" and "Toffoli" operations in academic literature, but are identified in Q# primarily as `CSWAP` and `CCNOT`.
+In both cases, the API documentation comments provide synonymous names based on proper nouns, along with all appropriate citations.
+
+This preference is especially important given that some usage of proper nouns will always be necessary — Q# follows the tradition set by many classical languages, for instance, and refers to `Bool` types in reference to Boolean logic, which is in turn named in honor of George Boole.
+A few quantum concepts similarly are named in a similar fashion, including the `Pauli` type built-in to the Q# language.
+By minimizing the usage of proper nouns where such usage is not essential, we reduce the impact where proper nouns cannot be reasonably avoided.
+
+# [Guidance](#tab/guidance) 
+
+We suggest:
+
+- Avoid the use of proper nouns in names.
+
+# [Examples](#tab/examples)
+
+***
+
+### Type Conversions ###
```

Comment:
Do you want to write some guideline regarding how to structure libraries? I was thinking something along the lines of conversion functions should be written when … and should go into e.g. a separate file/namespace under the name of ..., and should be aggregated e.g. by user defined type/ compilation unit etc.  Stuff of that sort.

Reply:
I think that for the most part, that's out of scope for a language style guide. Definitely something we should document somewhere, though. I'll make an issue to add that to the contribution guide.